flink-runtime/src/main/java/org/apache/flink/runtime/state/DirectoryStateHandle.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state;
+
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+
+import javax.annotation.Nonnull;
+
+import java.io.IOException;
+
+/**
+ * This state handle represents a directory. This class is, for example, used to represent the directory of RocksDB's
+ * native checkpoint directories for local recovery.
+ */
+public class DirectoryStateHandle implements StateObject {
+
+	/** Serial version. */
+	private static final long serialVersionUID = 1L;
+
+	/** The path that describes the directory. */
+	@Nonnull
+	private final Path directory;
+
+	/** The filesystem that contains the directory described by path. */
+	@Nonnull
+	private final FileSystem fileSystem;
```

Comment:
`FileSystem` is not serializable and should be removed from the `DirectoryStateHandle`. `directory` should give you all you need.

Reply:
It is true, and just was no problem because this handle is never serialized, it just is forced to inherit this from `StateObject`. 👍

Reply:
👍